### DISCUSSION TITLE
new supporter feature: auto download maps

### DISCUSSION THREAD

**Discussion by ackhack** [ppy/osu#13074] [Ideas]
**Describe the new feature:** 
Auto download maps by genre/artist/star-rating etc.

**Proposal designs of the feature:** 
Since everyone gets osu!direct, it would be an interesting idea to get a feature that auto downlaods maps with certain attributes, like stars>6.5 or artist=gyze etc.
Downloading them when the game starts would be enough.

**Comment by TreeCube** [comment]
This proposal wouldn’t work well due to possible storage restrictions. It would be a better idea if it was an option or button in the setting or on direct. The button would ask for a map amount goal and then ask for a maximum storage usage. This option would likely require someone to create an interface to select the types of maps. Otherwise, I think it would work well!

**Comment by ackhack** [comment]
or maybe the ability to filter the direct section like the beatmap selector and a download all visible Maps button.

**Comment by TreeCube** [comment]
A “cancel all” button would need to be added if there isn’t already one. Undoubtedly, people are gonna press the wrong thing and can’t stop all the downloads.

**Comment by ackhack** [comment]
Start to work on this in [this Branch](https://github.com/ackhack/osu/tree/beatmap-downloader). 

This Beatmap Downloader can automatically download Beatmaps filtered by

SR
Ranked/Loved/Both
Ruleset
It downloads Beatmaps until it hits the Date it last downloaded Beatmaps, so you will always have the newest Beatmaps downloaded. Default Date is DateTime.Now.

The Autodownload at the launch of the Game is configurable. Default is off.

Will remove the "Set Time to now" and "Set Time to 1 week ago" Buttons before merging, they are still there for manual testing if needed.
